### ISSUE TITLE
longevity 5000 tables in on keyspace, streaming failed (Cannot assign requested address)

### ISSUE THREAD

**Issue by fruch** [scylladb/scylla#4943]
This is Scylla's bug tracker, to be used for reporting bugs only.
If you have a question about Scylla, and not a bug, please ask it in
our mailing-list at scylladb-dev@googlegroups.com or in our slack channel.

- [x] I have read the disclaimer above, and I am reporting a suspected malfunction in Scylla.

## Installation details
Scylla version (or git commit hash): 3.1.0.rc4-0.20190830.d70c2db09
Cluster size: 2
OS (RHEL/CentOS/Ubuntu/AWS AMI):  ami-04b9560705a3975d1

## logs
gs://scratch.scylladb.com/ifruchte/5000ks_stream_error_node1.log
gs://scratch.scylladb.com/ifruchte/5000ks_stream_error_node2.log

## summery

after creating 5000 tables in one keyspace on node1, 
node2 two was added to the cluster and started to stream from node1.

node failed on stream in keyspace=feeds
**node1**:
```
2019-09-02T13:43:53+00:00  ip-10-0-136-244 !WARNING | scylla: [shard 0] stream_session - [Stream #aed42ad0-cd87-11e9-8cb8-000000000005] stream_transfer_task: Fail to send to 10.0.68.176:0: std::system_error (error system:99, connect: Cannot assign requested address)
2019-09-02T13:43:53+00:00  ip-10-0-136-244 !WARNING | scylla: [shard 0] stream_session - [Stream #aed42ad0-cd87-11e9-8cb8-000000000005] Failed to send: std::system_error (error system:99, connect: Cannot assign requested address)
2019-09-02T13:43:53+00:00  ip-10-0-136-244 !WARNING | scylla: [shard 0] stream_session - [Stream #aed42ad0-cd87-11e9-8cb8-000000000005] Streaming error occurred, peer=10.0.68.176
2019-09-02T13:43:53+00:00  ip-10-0-136-244 !WARNING | scylla: [shard 0] stream_session - [Stream #aed42ad0-cd87-11e9-8cb8-000000000005] Streaming plan for Bootstrap-feeds-index-0 failed, peers={10.0.68.176}, tx=0 KiB, 0.00 KiB/s, rx=0 KiB, 0.00 KiB/s
````
**node2**:
```
2019-09-02T13:35:02+00:00  ip-10-0-68-176 !INFO    | scylla: [shard 0] range_streamer - Bootstrap with 10.0.136.244 for keyspace=feeds, 51 out of 513 ranges: ranges = 51
2019-09-02T13:35:02+00:00  ip-10-0-68-176 !INFO    | scylla: [shard 0] stream_session - [Stream #773da070-cd86-11e9-8cb8-000000000005] Executing streaming plan for Bootstrap-feeds-index-0 with peers={10.0.136.244}, master
2019-09-02T13:35:10+00:00  ip-10-0-68-176 !INFO    | scylla: [shard 0] stream_session - [Stream #773da070-cd86-11e9-8cb8-000000000005] Received failed complete message, peer=10.0.136.244
2019-09-02T13:35:10+00:00  ip-10-0-68-176 !WARNING | scylla: [shard 0] stream_session - [Stream #773da070-cd86-11e9-8cb8-000000000005] Streaming plan for Bootstrap-feeds-index-0 failed, peers={10.0.136.244}, tx=0 KiB, 0.00 KiB/s, rx=0 KiB, 0.00 KiB/s
2019-09-02T13:35:10+00:00  ip-10-0-68-176 !WARNING | scylla: [shard 0] range_streamer - Bootstrap with 10.0.136.244 for keyspace=feeds failed, took 8.11 seconds: streaming::stream_exception (Stream failed)
2019-09-02T13:35:10+00:00  ip-10-0-68-176 !WARNING | scylla: [shard 0] range_streamer - Bootstrap failed, took 8 seconds, nr_ranges_remaining=513
2019-09-02T13:35:10+00:00  ip-10-0-68-176 !WARNING | scylla: [shard 0] range_streamer - Bootstrap failed to stream. Will retry in 60 seconds ...
```

nodes 2 retries again, but keep failing.

**Comment by slivne** [comment]
is this still up

can you run netstat -an on both machines

**Comment by fruch** [comment]
@slivne happen again on a new cluster,

I assumed you care only on the tcp

**node1**
```
[centos@ip-10-0-203-102 ~]$ netstat -an                                                                                                                                                                        
Active Internet connections (servers and established)                                                                                                                                                          
Proto Recv-Q Send-Q Local Address           Foreign Address         State                                                                                                                                      
tcp        0      0 0.0.0.0:111             0.0.0.0:*               LISTEN                                                                                                                                     
tcp        0      0 127.0.0.1:10000         0.0.0.0:*               LISTEN                                                                                                                                     
tcp        0      0 0.0.0.0:9042            0.0.0.0:*               LISTEN                                                                                                                                     
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN                                                                                                                                     
tcp        0      0 10.0.203.102:7000       0.0.0.0:*               LISTEN                                                                                                                                     
tcp        0      0 127.0.0.1:25            0.0.0.0:*               LISTEN                                                                                                                                     
tcp        0      0 0.0.0.0:9180            0.0.0.0:*               LISTEN                                                                                                                                     
tcp        0      0 0.0.0.0:9160            0.0.0.0:*               LISTEN                                                                                                                                     
tcp        0      0 10.0.203.102:9180       10.0.79.96:48276        ESTABLISHED                                                                                                                                
tcp        0    208 10.0.203.102:22         199.203.229.89:43580    ESTABLISHED                                                                                                                                
tcp        1      0 10.0.203.102:62608      169.254.169.254:80      CLOSE_WAIT                                                                                                                                 
tcp6       0      0 :::45839                :::*                    LISTEN                                                                                                                                     
tcp6       0      0 :::111                  :::*                    LISTEN                                                                                                                                     
tcp6       0      0 :::22                   :::*                    LISTEN                                                                                                                                     
tcp6       0      0 ::1:25                  :::*                    LISTEN                                                                                                                                     
tcp6       0      0 127.0.0.1:7199          :::*                    LISTEN                                                                                                                                     
tcp6       0      0 :::9100                 :::*                    LISTEN                                                                                                                                     
tcp6       0      0 127.0.0.1:52484         127.0.0.1:10000         TIME_WAIT                                                                                                                                  
tcp6       0      0 10.0.203.102:9100       10.0.79.96:45300        ESTABLISHED                                                                                                                                
udp        0      0 0.0.0.0:68              0.0.0.0:*                                                                                                                                                          
udp        0      0 0.0.0.0:111             0.0.0.0:*                                                                                                                                                          
udp        0      0 127.0.0.1:323           0.0.0.0:*                                                                                                                                                          
udp6       0      0 :::111                  :::*                                                                                                                                                               
udp6       0      0 ::1:323                 :::*                                               
```

**node2**
```
[centos@ip-10-0-176-25 ~]$ netstat -an                                                                                                                                                                         
Active Internet connections (servers and established)                                                                                                                                                          
Proto Recv-Q Send-Q Local Address           Foreign Address         State                                                                                                                                      
tcp        0      0 127.0.0.1:25            0.0.0.0:*               LISTEN                                                                                                                                     
tcp        0      0 0.0.0.0:111             0.0.0.0:*               LISTEN                                                                                                                                     
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN                                                                                                                                     
tcp        0    208 10.0.176.25:22          199.203.229.89:38208    ESTABLISHED                                                                                                                                
tcp6       0      0 ::1:25                  :::*                    LISTEN                                                                                                                                     
tcp6       0      0 127.0.0.1:7199          :::*                    LISTEN                                                                                                                                     
tcp6       0      0 :::9100                 :::*                    LISTEN                                                                                                                                     
tcp6       0      0 :::36975                :::*                    LISTEN                                                                                                                                     
tcp6       0      0 :::111                  :::*                    LISTEN                                                                                                                                     
tcp6       0      0 :::22                   :::*                    LISTEN                                                                                                                                     
tcp6       0      0 10.0.176.25:9100        10.0.79.96:46768        ESTABLISHED                                                                                                                                
udp        0      0 0.0.0.0:68              0.0.0.0:*                                                                                                                                                          
udp        0      0 0.0.0.0:111             0.0.0.0:*                                                                                                                                                          
udp        0      0 127.0.0.1:323           0.0.0.0:*                                                                                                                                                          
udp6       0      0 :::111                  :::*                                                                                                                                                               
udp6       0      0 ::1:323                 :::*                                
```

**Comment by gleb-cloudius** [comment]
On Mon, Sep 02, 2019 at 10:37:07PM -0700, Israel Fruchter wrote:
> @slivne happen again on a new cluster,
> 
> I assumed you care only on the tcp
> 
The netstat was taken long after the problem occurred. It is of no use.

--
			Gleb.

**Comment by fruch** [comment]
following @gleb-cloudius advice

I've restart the failing node (node2), and run the following on node1 while the problem happening
```
[centos@ip-10-0-203-102 ~]$ netstat -an  > netstat.log
[centos@ip-10-0-203-102 ~]$ vi netstat.log
[centos@ip-10-0-203-102 ~]$ grep TIME_WAIT  netstat.log   |  wc -l
15069
```
there are more then 15k line like that:
```
tcp        0      0 10.0.203.102:55892      10.0.176.25:7000        TIME_WAIT
```
the full netstat log:
http://scratch.scylladb.com/fruch/4943_netstat.log

**Comment by gleb-cloudius** [comment]
@asias why do we open so many streams? Didn't you fix it?

**Comment by slivne** [comment]
@asias ping

**Comment by slivne** [comment]
asias's patch for not opening connections on all cores 0e6b62244c1d658ddd0a733d81033eab7f361e65 is merged into 3.1 - so we are not missing this backport

**Comment by asias** [comment]
From start to bootstrap node2 until the first failure of streaming on node1 (the sender)

There are only 2684 stream plan connects from node1 to node2 on all shards, 1452 out of the 2684 are the connections for the keyspace feeds which has 5000 tables. 

[asias@hjpc2 issue.4943]$ cat stream.first.txt |grep 'Start send'|wc -l
2684
[asias@hjpc2 issue.4943]$ cat stream.first.txt |grep 'Start send'|grep ks=feeds|wc -l
1452

This is not supposed to exhaust the ports on node1 by itself. There must be something else that is not streaming to use the ports, prometheus server?

**Comment by asias** [comment]
I created https://github.com/scylladb/scylla/pull/4968 to open the connection if there is data in the reader, in addition to what we had in the past: to open the connection if the range is relevant for the shard. 

But I think we have other issues that exhaust the ports beyond streaming.

**Comment by slivne** [comment]
@slivne try to reproduce outside of sct

**Comment by slivne** [comment]
easy way to reproduce 

create a single node with 500 empty tables using a large machine (i3.16x) 

[schema.txt](https://github.com/scylladb/scylla/files/3581348/schema.txt)

add a new node / `nodetool rebuild` it and you will get

on 3.1.0.rc5

````
[shard 0] stream_session - [Stream #ae8b4c10-d026-11e9-b650-000000000009] Streaming plan for Rebuild-keyspace1-index-1 failed, peers={10.0.0.140}, tx=0 Ki
B, 0.00 KiB/s, rx=0 KiB, 0.00 KiB/s
[shard 0] stream_session - [Stream #ae8b4c10-d026-11e9-b650-000000000009] Streaming error occurred, peer=10.0.0.140
[shard 0] stream_session - [Stream #ae8b4c10-d026-11e9-b650-000000000009] Failed to send: std::system_error (error system:99, connect: Cannot assign requested address)
[shard 0] stream_session - [Stream #ae8b4c10-d026-11e9-b650-000000000009] stream_transfer_task: Fail to send to 10.0.0.140:0: std::system_error (error system:99, connect: Cannot assign requested address)
```

```
[centos@ip-10-0-0-68 ~]$ netstat -an | grep TIME_WAIT | wc -l
32697
```

on 3.0.10

```
Sep 05 21:52:51 ip-10-0-0-13.ec2.internal scylla[3277]:  [shard 4] stream_session - [Stream #7fc3c870-d027-11e9-81c0-000000000004] Streaming plan for Rebuild-keyspace1-index-0 failed, peers={10.0.0.160}, tx=0 KiB, 0.00 KiB/s, rx=0 KiB, 0.00 KiB/s
Sep 05 21:52:51 ip-10-0-0-13.ec2.internal scylla[3277]:  [shard 4] stream_session - [Stream #7fc3c870-d027-11e9-81c0-000000000004] Streaming error occurred, peer=10.0.0.160
Sep 05 21:52:51 ip-10-0-0-13.ec2.internal scylla[3277]:  [shard 4] stream_session - [Stream #7fc3c870-d027-11e9-81c0-000000000004] Failed to send: std::system_error (error system:98, Address already in use)
Sep 05 21:52:51 ip-10-0-0-13.ec2.internal scylla[3277]:  [shard 4] stream_session - [Stream #7fc3c870-d027-11e9-81c0-000000000004] stream_transfer_task: Fail to send to 10.0.0.160:0: std::system_error (error system:98, Address already in use)
```

```
[centos@ip-10-0-0-13 ~]$ netstat -an | grep TIME | wc -l
28102
```

so this is related to the new streaming that opens connections for every token range - once its closed - this is kept around in TIME_WAIT state and as we can see it causes issue as we run out of ports we have been discussing reusing streaming connections and not disposing them in the rpc layer - not sure we have a real choice but togo ahead and do that.

@gleb-cloudius / @avikivity  / @asias - thoughts ?

@asias

**Comment by slivne** [comment]
@roydahan / @fruch did this really work for you guys on 3.0.X ?

**Comment by fruch** [comment]
We run it on 3.0.8, and it did pass that stage. We managed to add 5 nodes.

**Comment by slivne** [comment]
Well indeed it seems that bootstrap:
* on 3.0.10 ScyllaDB 3.0.10 (ami-0b88175d56d413224) the bootstrap failes 
```   Failed to send: std::system_error (error system:98, Address already in use) ```
* on 3.0.9 ScyllaDB 3.0.9 (ami-08e617d9f57f477ee) the bootstrap works
* on 3.0.8  ScyllaDB 3.0.8 (ami-08d3b18bed79ebefc) the bootstrap works 

So we have a point to start the search the diff between 3.0.9 ... 3.0.10

The most probable (in my view is cc0b4d249b95aca0d69966e8a29b2a7927869653)
@gleb-cloudius I'll need your help with this Asias is out this week 

how to reproduce

1. Boot first node - node1  i3.16X
2. node1 run cqlsh -f schema.txt
[schema.txt](https://github.com/scylladb/scylla/files/3587020/schema.txt)
3. Boot second node - node2  i3.16x
4. node2 ``` sudo service scylla-server stop```
5. node2 ```sudo rm -Rf /var//lib/scylla/data/* /var//lib/scylla/commitlog/*
6. node2 - change seed to node1
7. node2 ``` sudo service scylla-server start

**Comment by gleb-cloudius** [comment]
@slivne can you try those patches:


[seastar.txt](https://github.com/scylladb/scylla/files/3587625/seastar.txt)
[scylla.txt](https://github.com/scylladb/scylla/files/3587626/scylla.txt)

**Comment by slivne** [comment]
tried branches are on 
seastar-dev/shlomi/gleb-4943-3-1
seastar-dev/shlomi/gleb-4943-seastar-3-1

rpms wget http://scratch.scylladb.com/shlomi/scylla-3.1.0-gleb-4943.tar

I installed on topof 3.1.0.rc5 AMI
getting

```
Sep 08 20:44:29 ip-10-0-0-90.ec2.internal scylla[15244]:  [shard 0] stream_session - [Stream #74128f80-d279-11e9-a46a-00000000001a] Failed to send: std::system_error (error system:99, connect: Cannot assign requested address)
```
```
 netstat -an | grep TIME | wc -l
32693
```

**Comment by gleb-cloudius** [comment]
On Sun, Sep 08, 2019 at 01:48:58PM -0700, Shlomi Livne wrote:
> tried branches are on 
> seastar-dev/shlomi/gleb-4943-3-1
> seastar-dev/shlomi/gleb-4943-seastar-3-1
> 
> rpms wget http://scratch.scylladb.com/shlomi/scylla-3.1.0-gleb-4943.tar
> 
> I installed on topof 3.1.0.rc5 AMI
> getting
> 
> ```
> Sep 08 20:44:29 ip-10-0-0-90.ec2.internal scylla[15244]:  [shard 0] stream_session - [Stream #74128f80-d279-11e9-a46a-00000000001a] Failed to send: std::system_error (error system:99, connect: Cannot assign requested address)
> ```
> ```
>  netstat -an | grep TIME | wc -l
> 32693
> ```
> 
The patch was not suppose to eliminate TIME_WAIT state, it supposed to
allow to reuse addresses with this state. So either the path is
incorrect, SO_REUSEADDRESS does not works for client sockets or those
sockets are not in TIME_WAIT state when we try to connect and become
such only later.

--
			Gleb.

**Comment by gleb-cloudius** [comment]
It looks like previous patch sets reuseaddr option on a wrong socket. Can you try this one:

[seastar2.txt](https://github.com/scylladb/scylla/files/3590323/seastar2.txt)

**Comment by slivne** [comment]
Installed a new version with this 

tried branches are on
seastar-dev/shlomi/gleb-4943-3-1
seastar-dev/shlomi/gleb-4943-seastar-3-1_v2

rpms wget http://scratch.scylladb.com/shlomi/scylla-3.1.0-gleb-4943_v2.tar


The same with this version 

```
Sep 10 07:40:49 ip-10-0-0-90.ec2.internal scylla[64351]:  [shard 0] stream_session - [Stream #4f36b670-d39e-11e9-b3d2-00000000000f] Failed to send: std::system_error (error system:99, connect: Cannot assign requested address)
```

**Comment by slivne** [comment]
yet it seems that SO_REUSEADDR was set ... the number of ports in TIME_WAIT is much lower
```
[centos@ip-10-0-0-90 ~]$ netstat -an | grep TIME | wc -l
61
```

so something else is broken here ...  @gleb-cloudius

**Comment by gleb-cloudius** [comment]
Can you try with this patch on top of previous one?

[eaddrnoavail.txt](https://github.com/scylladb/scylla/files/3594718/eaddrnoavail.txt)

**Comment by gleb-cloudius** [comment]
One more on top:

[recreate-socket.txt](https://github.com/scylladb/scylla/files/3595761/recreate-socket.txt)

**Comment by slivne** [comment]
with the last I got 

```
Sep 10 14:27:41 ip-10-0-0-183.ec2.internal scylla[5638]:  [shard 31] rpc - client 10.0.0.90:56575: server connection dropped: connection is closed
Sep 10 14:27:41 ip-10-0-0-183.ec2.internal scylla[5638]:  [shard 0] stream_session - [Stream #255270e0-d3d7-11e9-b8cb-00000000000f] Received failed complete message, peer=10.0.0.90
Sep 10 14:27:41 ip-10-0-0-183.ec2.internal scylla[5638]:  [shard 0] stream_session - [Stream #255270e0-d3d7-11e9-b8cb-00000000000f] Streaming plan for Bootstrap-keyspace1-index-0 failed, peers={10.0.0.90}, tx=0 KiB, 0.00 KiB/s, rx=0 KiB, 0.00 KiB/s
Sep 10 14:27:41 ip-10-0-0-183.ec2.internal scylla[5638]:  [shard 0] range_streamer - Bootstrap with 10.0.0.90 for keyspace=keyspace1 failed, took 0.977 seconds: streaming::stream_exception (Stream failed)
Sep 10 14:27:41 ip-10-0-0-183.ec2.internal scylla[5638]:  [shard 0] range_streamer - Bootstrap failed, took 0 seconds, nr_ranges_remaining=513
Sep 10 14:27:41 ip-10-0-0-183.ec2.internal scylla[5638]:  [shard 0] range_streamer - Bootstrap failed to stream. Will retry in 202 seconds ...

```

and 

```
Sep 10 14:27:41 ip-10-0-0-90.ec2.internal scylla[710]:  [shard 21] stream_session - [Stream #255270e0-d3d7-11e9-b8cb-00000000000f] Start sending ks=keyspace1, cf=standard_1_43, estimated_partitions=0, with new rpc streaming
Sep 10 14:27:41 ip-10-0-0-90.ec2.internal scylla[710]:  [shard 0] stream_session - [Stream #255270e0-d3d7-11e9-b8cb-00000000000f] stream_transfer_task: Fail to send to 10.0.0.183:0: std::system_error (error system:107, read: Transport endpoint is not connected)
Sep 10 14:27:41 ip-10-0-0-90.ec2.internal scylla[710]:  [shard 0] stream_session - [Stream #255270e0-d3d7-11e9-b8cb-00000000000f] Failed to send: std::system_error (error system:107, read: Transport endpoint is not connected)
Sep 10 14:27:41 ip-10-0-0-90.ec2.internal scylla[710]:  [shard 0] stream_session - [Stream #255270e0-d3d7-11e9-b8cb-00000000000f] Streaming error occurred, peer=10.0.0.183
Sep 10 14:27:41 ip-10-0-0-90.ec2.internal scylla[710]:  [shard 0] stream_session - [Stream #255270e0-d3d7-11e9-b8cb-00000000000f] Streaming plan for Bootstrap-keyspace1-index-0 failed, peers={10.0.0.183}, tx=0 KiB, 0.00 KiB/s, rx=0 KiB, 0.00 KiB/s
^C
```

**Comment by slivne** [comment]
The latest versions are on

tried branches are on
seastar-dev/shlomi/gleb-4943-3-1
seastar-dev/shlomi/gleb-4943-seastar-3-1_v5


the errors I get with these version are 

seed node
```
Sep 12 07:22:43 ip-10-0-0-100.ec2.internal scylla[78515]:  [shard 1] stream_session - [Stream #1c38b0c0-d52e-11e9-816f-00000000000e] Start sending ks=system_traces, cf=node_slow_log_time_idx, estimated_partition
Sep 12 07:22:43 ip-10-0-0-100.ec2.internal scylla[78515]:  [shard 38] rpc - client 10.0.0.119:7000: client stream connection dropped: read: Transport endpoint is not connected
Sep 12 07:22:43 ip-10-0-0-100.ec2.internal scylla[78515]:  [shard 0] stream_session - [Stream #1c38b0c0-d52e-11e9-816f-00000000000e] stream_transfer_task: Fail to send to 10.0.0.119:0: std::system_error (error s
Sep 12 07:22:43 ip-10-0-0-100.ec2.internal scylla[78515]:  [shard 0] stream_session - [Stream #1c38b0c0-d52e-11e9-816f-00000000000e] Failed to send: std::system_error (error system:107, read: Transport endpoint 
Sep 12 07:22:43 ip-10-0-0-100.ec2.internal scylla[78515]:  [shard 0] stream_session - [Stream #1c38b0c0-d52e-11e9-816f-00000000000e] Streaming error occurred, peer=10.0.0.119
Sep 12 07:22:43 ip-10-0-0-100.ec2.internal scylla[78515]:  [shard 0] stream_session - [Stream #1c38b0c0-d52e-11e9-816f-00000000000e] Streaming plan for Bootstrap-system_traces-index-6 failed, peers={10.0.0.119},
Sep 12 07:27:47 ip-10-0-0-100.ec2.internal scylla[78515]:  [shard 0] rpc - client 10.0.0.119:7000: fail to connect: Connection refused

```

bootstrapping node
```
Sep 12 07:22:43 ip-10-0-0-119.ec2.internal scylla[71491]:  [shard 0] stream_session - [Stream #1c38b0c0-d52e-11e9-816f-00000000000e] Executing streaming plan for Bootstrap-system_traces-index-6 with peers={10.0.
Sep 12 07:22:43 ip-10-0-0-119.ec2.internal scylla[71491]:  [shard 38] rpc - client 10.0.0.100:57450: server connection dropped: connection is closed
Sep 12 07:22:43 ip-10-0-0-119.ec2.internal scylla[71491]:  [shard 0] stream_session - [Stream #1c38b0c0-d52e-11e9-816f-00000000000e] Received failed complete message, peer=10.0.0.100
Sep 12 07:22:43 ip-10-0-0-119.ec2.internal scylla[71491]:  [shard 0] stream_session - [Stream #1c38b0c0-d52e-11e9-816f-00000000000e] Streaming plan for Bootstrap-system_traces-index-6 failed, peers={10.0.0.100},
Sep 12 07:22:43 ip-10-0-0-119.ec2.internal scylla[71491]:  [shard 0] range_streamer - Bootstrap with 10.0.0.100 for keyspace=system_traces failed, took 0.596 seconds: streaming::stream_exception (Stream failed)
Sep 12 07:22:43 ip-10-0-0-119.ec2.internal scylla[71491]:  [shard 0] range_streamer - Bootstrap failed, took 0 seconds, nr_ranges_remaining=1041
Sep 12 07:22:43 ip-10-0-0-119.ec2.internal scylla[71491]:  [shard 0] range_streamer - Bootstrap failed to stream. Will retry in 303 seconds ...
Sep 12 07:27:46 ip-10-0-0-119.ec2.internal scylla[71491]:  [shard 0] boot_strapper - Error during bootstrap: streaming::stream_exception (Stream failed)
```
attached are the tcp dums

[bootstrap.cap.zip](https://github.com/scylladb/scylla/files/3604278/bootstrap.cap.zip)
[seed.cap.zip](https://github.com/scylladb/scylla/files/3604279/seed.cap.zip)

**Comment by slivne** [comment]
@gleb-cloudius please update the issue with the latest info

**Comment by bhalevy** [comment]
@roydahan : @fruch is testing the fix.
@avikivity will backport 73e3d0a2832b0d9cf626f70b9d7fe3ff41e47769 once merged to master.

**Comment by fruch** [comment]
@bhalevy I would test, once all the fix is merged, only the seastar portion of it was merged. waiting for scylla part to get into master

**Comment by bhalevy** [comment]
@fruch 73e3d0a2832b0d9cf626f70b9d7fe3ff41e47769 was merged to master, do you need it backported for starting to test or can you test it provisionally?

**Comment by fruch** [comment]
I can try the master AMI, if this is needed to be backport to 3.1, that's you call.

**Comment by bhalevy** [comment]
@avikivity please backport 73e3d0a2832b0d9cf626f70b9d7fe3ff41e47769 to 3.1

**Comment by fruch** [comment]
I've rerun this test case with the master ami, failed with coredump on a different issue but this seem to be fixed.

**Comment by avikivity** [comment]
I'd like to understand the regression status of this bug. Is it a regression in 3.0? If so which commit regressed it? Is it a regression in 3.1 relative to 3.0?

**Comment by slivne** [comment]
@gleb-cloudius / @asias can you help

 AFAIK in 3.0 - this is a regression the source of this is the change that was releases as part of 3.0.9 cc0b4d249b95aca0d69966e8a29b2a7927869653

the fix from gleb was 
"      > Merge "fix some tcp connection bugs and add reuseaddr option to a client socket" from Gleb"

**Comment by avikivity** [comment]
Already backported to all vulnerable branches, removing "Backport candidate" label.

Note: "streaming: Do not open rpc stream connection if reader has no data" was not backported. @asias
